### PR TITLE
app: add beacon node peer count metric

### PR DESCRIPTION
**Pull request by xenowits**
Adds beacon node peer count metric gauge.

category: feature
ticket: #1603

### CODE REVIEW

File: app/monitoringapi.go [hunk 1]
```diff
@@ -186,6 +189,35 @@ func beaconNodeSyncing(ctx context.Context, eth2Cl eth2client.NodeSyncingProvide
 	return state.IsSyncing, nil
 }
 
+// peerCounter populates the peerCountGauge with the beacon node peer count.
+func peerCounter(ctx context.Context, eth2Cl eth2wrap.Client, clock clockwork.Clock) {
+	ticker := clock.NewTicker(10 * time.Second)
```
Comment (xenowits) [inline app/monitoringapi.go:194]: please suggest query interval if this is not ok

File: testutil/beaconmock/options.go [hunk 2]
```diff
@@ -489,6 +491,10 @@ func defaultMock(httpMock HTTPMock, httpServer *http.Server, clock clockwork.Clo
 		BlockAttestationsFunc: func(ctx context.Context, stateID string) ([]*eth2p0.Attestation, error) {
 			return []*eth2p0.Attestation{}, nil
 		},
+		NodePeerCountFunc: func(ctx context.Context) (eth2wrap.PeerCount, error) {
+			//nolint:gosec
+			return eth2wrap.PeerCount{Connected: rand.Int() % 100}, nil
```
Comment (xenowits) [inline testutil/beaconmock/options.go:496]: i'm open to this being constant

File: app/monitoringapi.go [hunk 3]
```diff
@@ -186,6 +189,35 @@ func beaconNodeSyncing(ctx context.Context, eth2Cl eth2client.NodeSyncingProvide
...
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-ticker.Chan():
```
Comment (corverroos) [inline app/monitoringapi.go:200]: also do this on startup

File: app/monitoringapi.go [hunk 4]
```diff
@@ -186,6 +189,35 @@ func beaconNodeSyncing(ctx context.Context, eth2Cl eth2client.NodeSyncingProvide
...
+				resp, err := beaconNodePeerCount(ctx, eth2Cl)
+				if err != nil {
+					log.Error(ctx, "Failed to get beacon node peer count", err)
+				}
+				peerCountGauge.Set(float64(resp))
```
Comment (corverroos) [inline app/monitoringapi.go:205]: do not set the gauge if the query failed.

File: app/monitoringapi.go [hunk 5]
```diff
@@ -186,6 +189,35 @@ func beaconNodeSyncing(ctx context.Context, eth2Cl eth2client.NodeSyncingProvide
...
+// peerCounter populates the peerCountGauge with the beacon node peer count.
+func peerCounter(ctx context.Context, eth2Cl eth2wrap.Client, clock clockwork.Clock) {
+	ticker := clock.NewTicker(10 * time.Second)
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-ticker.Chan():
+				resp, err := beaconNodePeerCount(ctx, eth2Cl)
```
Comment (corverroos) [inline app/monitoringapi.go:201]: rename `resp` to `count`

File: testutil/beaconmock/options.go (hunk 2 above)
Comment (corverroos) [reply testutil/beaconmock/options.go:496]: yes please, zero `eth2wrap.PeerCount` struct.

**Review by corverroos — APPROVED** [review]

File: app/monitoringapi.go (hunk 3 above)
Comment (xenowits) [reply app/monitoringapi.go:200]: cool, will use your pattern:
```go
		onStartup := make(chan struct{}, 1)
		onStartup <- struct{}{}
```

**Pull request merged**